File: app/src/main/java/dev/ragnarok/fenrir/api/model/VKApiStickerSet.java
package dev.ragnarok.fenrir.api.model;

import com.google.gson.annotations.SerializedName;

import java.util.List;

public class VKApiStickerSet implements Identificable {
    @SerializedName("background")
    public String background;

    @SerializedName("description")
    public String description;

    @SerializedName("author")
    public String author;

    @SerializedName("free")
    public boolean free;

    @SerializedName("can_purchase")
    public boolean can_purchase;

    @SerializedName("payment_type")
    public String payment_type;

    @SerializedName("product")
    public Product product;

    @Override
    public int getId() {
        return product.id;
    }

    public static final class Image {
        @SerializedName("url")
        public String url;

        @SerializedName("width")
        public int width;

        @SerializedName("height")
        public int height;
    }

    public static class Product {

        @SerializedName("id")
        public int id;

        @SerializedName("purchased")
        public boolean purchased;

        @SerializedName("title")
        public String title;

        @SerializedName("promoted")
        public boolean promoted;

        @SerializedName("active")
        public boolean active;

        @SerializedName("type")
        public String type;

        @SerializedName("icon")
        public List<Image> icon;

        @SerializedName("stickers")
        public List<VKApiSticker> stickers;
    }
}

File: app/src/main/kotlin/dev/ragnarok/fenrir/api/model/VKApiStickerSet.kt
package dev.ragnarok.fenrir.api.model

import dev.ragnarok.fenrir.orZero
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
class VKApiStickerSet : Identificable {
    @SerialName("background")
    var background: String? = null

    @SerialName("description")
    var description: String? = null

    @SerialName("author")
    var author: String? = null

    @SerialName("free")
    var free = false

    @SerialName("can_purchase")
    var can_purchase = false

    @SerialName("payment_type")
    var payment_type: String? = null

    @SerialName("product")
    var product: Product? = null
    override fun getObjectId(): Int {
        return product?.id.orZero()
    }

    @Serializable
    class Image {
        @SerialName("url")
        var url: String? = null

        @SerialName("width")
        var width = 0

        @SerialName("height")
        var height = 0
    }

    @Serializable
    class Product {
        @SerialName("id")
        var id = 0

        @SerialName("purchased")
        var purchased = false

        @SerialName("title")
        var title: String? = null

        @SerialName("promoted")
        var promoted = false

        @SerialName("active")
        var active = false

        @SerialName("type")
        var type: String? = null

        @SerialName("icon")
        var icon: List<Image>? = null

        @SerialName("stickers")
        var stickers: List<VKApiSticker>? = null
    }
}